aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -0,0 +1,359 @@
+
+Register health check services with `AddHealthChecks` in `Startup.ConfigureServices`. The sample app adds the `SqlConnectionHealthCheck` with the `AddCheck` extension method (*DbHealthStartup.cs*):
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/DbHealthStartup.cs?name=snippet_ConfigureServices)]
+
+Call Health Check Middleware in the app processing pipeline in `Startup.Configure`:
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/DbHealthStartup.cs?name=snippet_Configure&highlight=3)]
+
+To run the database probe scenario using the sample app, execute the following command from the project's folder in a command shell:
+
+```console
+dotnet run --scenario db
+```
+
+## Separate readiness and liveness probes
+
+In some hosting scenarios, a pair of health checks are used that distinguish two app states:
+
+* The app is functioning but not yet ready to receive requests. This state is the app's *readiness*.
+* The app is functioning and responding to requests. This state is the app's *liveness*.
+
+The readiness check usually performs a more extensive and time-consuming set of checks to determine if all of the app's subsystems and resources are available. A liveness check merely performs a quick check to determine if the app is available to process requests. After the app passes its readiness check, there's no need to burden the app further with the expensive set of readiness checks&mdash;further checks only require checking for liveness.
+
+The sample app includes a `SlowDependencyHealthCheck` that creates a contrived 15 second delay when the check is executed. During the delay, `HealthCheckResult.Unhealthy` is returned by the readiness check. After the delay, `HealthCheckResult.Healthy` is returned (*SlowDependencyHealthCheck.cs*):
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/SlowDependencyHealthCheck.cs?name=snippet1&highlight=7,15-22)]
+
+Register health check services with `AddHealthChecks` in `Startup.ConfigureServices`. Register `SlowDependencyHealthCheck` with `AddCheck` (*LivenessProbeStartup.cs*):
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/LivenessProbeStartup.cs?name=snippet_ConfigureServices)]
+
+Call Health Check Middleware in the app processing pipeline in `Startup.Configure`. In the sample app, the health check endpoints are created at `/health/ready` for the readiness check and `/health/live` for the liveness check. The liveness check filters out the `SlowDependencyHealthCheck` by returning `false` in the `HealthCheckOptions.Predicate` (for more information, see [Filter health checks](#filter-health-checks)):
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/LivenessProbeStartup.cs?name=snippet_Configure&highlight=3,5-9)]
+
+To run the readiness/liveness configuration scenario using the sample app, execute the following command from the project's folder in a command shell:
+
+```console
+dotnet run --scenario liveness
+```
```

Comment:
This doesn't really change in preview3, except now we have tags. I'd show an example that uses tags for `liveness` and `readiness` to filter.